### PR TITLE
flannel: doc fix and check for missing subnet.env fields

### LINKED ISSUE #72: Missing FLANNEL_NETWORK in flannel documentation.
In https://github.com/appc/cni/blob/master/Documentation/flannel.md example `subnet.env` does not have `FLANNEL_NETWORK`  variable, without which https://github.com/appc/cni/blob/master/plugins/meta/flannel/flannel.go#L205 will panic.

Btw. there is no error checking if required variables are in `subnet.env` file.
If this variables could be missing, then in https://github.com/appc/cni/blob/master/plugins/meta/flannel/flannel.go#L72 pointer struct members should be initialized with empty values like:
```
se := &subnetEnv{
  nw: &net.IPNet{},
  sn: &net.IPNet{},
}
```

### PR DESCRIPTION
**Pull request by eyakubovich**
- Document that flannel outputs FLANNEL_NETWORK b/c we use it
- Check and error out if expecited fields are missing in subnetEnv file

Fixes #72

### CODE REVIEW

File: plugins/meta/flannel/flannel.go [hunk 1]
```diff
@@ -92,16 +108,22 @@ func loadFlannelSubnetEnv(fn string) (*subnetEnv, error) {
 			if err != nil {
 				return nil, err
 			}
-			se.mtu = uint(mtu)
+			se.mtu = new(uint)
```
Comment (jonboulle) [inline plugins/meta/flannel/flannel.go]: what's wrong with se.mtu = &mtu?

File: plugins/meta/flannel/flannel.go [hunk 2]
```diff
@@ -92,16 +108,22 @@ func loadFlannelSubnetEnv(fn string) (*subnetEnv, error) {
...
+			ipmasq := parts[1] == "true"
+			se.ipmasq = &ipmasq
 		}
 	}
 	if err := s.Err(); err != nil {
 		return nil, err
 	}
 
+	if m := se.missing(); m != "" {
+		return nil, fmt.Errorf("%q is missing %q", fn, m)
```
Comment (jonboulle) [inline plugins/meta/flannel/flannel.go]: you don't want to be nice and print all the missing ones? ;-)

File: plugins/meta/flannel/flannel.go (hunk 1 above)
Comment (eyakubovich) [inline plugins/meta/flannel/flannel.go]: you'll get a type mismatch. `mtu` is `uint64` and `se.mtu` is `*uint`. There are other ways to write this code, of course.

File: plugins/meta/flannel/flannel.go (hunk 2 above)
Comment (eyakubovich) [inline plugins/meta/flannel/flannel.go]: fine, since you ask :)

File: plugins/meta/flannel/flannel.go (hunk 1 above)
Comment (jonboulle) [inline plugins/meta/flannel/flannel.go]: oh ok clearly I didn't read the spec of ParseUint. *cringe* sometimes I miss Java's extreme type verbosity..

**Comment by eyakubovich** [comment]
@jonboulle PTAL

File: plugins/meta/flannel/flannel.go [hunk 3]
```diff
@@ -182,7 +206,7 @@ func cmdAdd(args *skel.CmdArgs) error {
 
 	if !hasKey(n.Delegate, "ipMasq") {
 		// if flannel is not doing ipmasq, we should
-		ipmasq := !fenv.ipmasq
+		ipmasq := !*fenv.ipmasq
```
Comment (jonboulle) [inline plugins/meta/flannel/flannel.go]: did I miss this change in the previous review? if not, did it actually compile!?

**Comment by jonboulle** [comment]
LGTM although I am slightly concerned about the above

File: plugins/meta/flannel/flannel.go (hunk 3 above)
Comment (eyakubovich) [inline plugins/meta/flannel/flannel.go]: I made all fields pointers so I can use `nil` as not set.

Comment (jonboulle) [inline plugins/meta/flannel/flannel.go]: I got that, I just didn't notice you change this line in the original
commit.

**Pull request merged**